src/base/bittorrent/session.cpp

```diff
@@ -3929,6 +3929,37 @@ quint64 Session::getAlltimeUL() const
     return m_statistics->getAlltimeUL();
 }
 
+QVariantMap Session::getStats() const
+{
+    const quint64 atd = getAlltimeDL();
+    const quint64 atu = getAlltimeUL();
+
+    // num_peers is not reliable (adds up peers, which didn't even overcome tcp handshake)
+    quint32 peers = 0;
+    foreach (BitTorrent::TorrentHandle *const torrent, torrents())
```

Comment:
For new code **always** use ranged-based for.
`for (const BitTorrent::TorrentHandle *torrent : torrents())`